be/src/exec/pipeline/hashjoin/spillable_hash_join_probe_operator.cpp

```diff
@@ -216,12 +205,14 @@ Status SpillableHashJoinProbeOperator::_push_probe_chunk(RuntimeState* state, co
     auto partition_processer = [&chunk, this, state, &hash_values](spill::SpilledPartition* probe_partition,
                                                                    const std::vector<uint32_t>& selection, int32_t from,
                                                                    int32_t size) {
-        // now spill only support inner join and left semi join
         // nothing to do for empty partition
-        auto build_partition_iter = _pid_to_build_partition.find(probe_partition->partition_id);
-        if (build_partition_iter != _pid_to_build_partition.end()) {
-            if (build_partition_iter->second->empty()) {
-                return;
+        if (is_left_semi_join_or_inner_join(_join_prober->hash_table_param().join_type)) {
+            // For left semi join and inner join we can just skip the empty partition
```

Comment:
why left anti and left outer join is excluded?

Reply:
it means if right partition is empty. we don't have to probe any data in the same partition. we don't have to probe any data. so left anti and left outer join should be excluded.

but it seems right semi/right anti/right outer join could short circuit. I will add it.
```
        if (row_count == 0 && (_join_type == TJoinOp::INNER_JOIN || _join_type == TJoinOp::LEFT_SEMI_JOIN ||
                               _join_type == TJoinOp::RIGHT_SEMI_JOIN || _join_type == TJoinOp::RIGHT_ANTI_JOIN ||
                               _join_type == TJoinOp::RIGHT_OUTER_JOIN)) {
            _phase = HashJoinPhase::EOS;
            return;
        }
```

Reply:
add a function could_short_circuit